global make install is broken both with and without sudo on MacOS, and should be tested on Coq's CI

Since the dune PR was merged, I have been unable to test fiat-crypto on MacOS CI.  The current error is
```
DUNE      _build/default/coq-core.install
    ocamlopt tools/coqworkmgr.exe
ld: warning: directory not found for option '-L/opt/local/lib'
File "_none_", line 1:
Error: Error on dynamically loaded library: dllzarith.so: dlopen(dllzarith.so, 10): image not found
make[1]: *** [_build/default/coq-core.install] Error 1
make: *** [submake] Error 2
```
This is very wonky, why does the install target try to open zarith?

I'm using the build script from Coq's CI, only changing the prefix to /use/local.  Full log at https://github.com/mit-plv/fiat-crypto/runs/2480414838?check_suite_focus=true

cc @ejgallego

--- a/Makefile.build
+++ b/Makefile.build
@@ -89,7 +89,7 @@ include Makefile.install
 
 world: coq coqide documentation revision
 byte: world
-coq: coqlib coqbinaries tools
+coq: coqlib coqbinaries tools $(BCONTEXT)/coq-core.install $(BCONTEXT)/coqide-server.install
 
 world.timing.diff: coq.timing.diff
 coq.timing.diff: coqlib.timing.diff

--- a/Makefile.ide
+++ b/Makefile.ide
@@ -22,7 +22,7 @@ IDEFILES=$(wildcard ide/*.lang) ide/coq_style.xml ide/coq.png) $(IDEBINDINGS)
 ifeq ($(HASCOQIDE),no)
 coqide:
 else
-coqide: $(COQIDE) $(IDETOP) $(TOPBINOPT) $(IDEBINDINGS) $(VO_OUT_DIR)theories/Init/Prelude.$(VO)
+coqide: $(COQIDE) $(IDETOP) $(TOPBINOPT) $(IDEBINDINGS) $(VO_OUT_DIR)theories/Init/Prelude.$(VO) $(BCONTEXT)/coqide.install
 endif
 
 # $(INSTALLLIB) ide/coq.png ide/*.lang ide/coq_style.xml $(IDEBINDINGS) $(FULLDATADIR)

--- a/Makefile.install
+++ b/Makefile.install
@@ -56,6 +56,13 @@ endif
 .PHONY: install-coq install-dune install-library
 
 # --display quiet should not produce output, dune bug
+
+# Note that dune install paths are setup at dune's build time, thus
+# for example a Debian-built dune will install stuff following
+# Debian's FHS, an opam-built dune will differ and follow OPAM's layout.
+#
+# We ought to improve this, as of today the case of an opam-built dune
+# installing to a global prefix such as /usr/local/ may not follow the FHS
 install-dune: $(BCONTEXT)/coq-core.install $(BCONTEXT)/coqide-server.install
 	dune install --display quiet --mandir="$(MANDIR)" --prefix="$(FULLCOQPREFIX)" coq-core
 	dune install --display quiet --mandir="$(MANDIR)" --prefix="$(FULLCOQPREFIX)" coqide-server

